src/components/Results/Summary.vue

```diff
@@ -0,0 +1,254 @@
+<!--
+  - @copyright Copyright (c) 2020 Jan C. Borchardt https://jancborchardt.net
+  -
+  - @author Jan C. Borchardt https://jancborchardt.net
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program.  If not, see <http://www.gnu.org/licenses/>.
+  -
+  -->
+
+<template>
+	<div class="section question-summary">
+		<h3>{{ question.text }}</h3>
+		<p class="question-summary__detail">
+			{{ answerTypes[question.type].label }}
+		</p>
+		<ol v-if="question.type == 'multiple' || question.type == 'multiple_unique' || question.type == 'dropdown'"
+			class="question-summary__statistic">
+			<li v-for="(questionOption, index) in questionOptions"
+				:key="index">
+				<label :for="questionOption.text.replace(/\s+/g, '')">
+					{{ questionOption.count }} <span class="question-summary__statistic-percentage">({{ questionOption.percentage }}%):</span>
+					<span class="question-summary__statistic-text">{{ questionOption.text }}</span>
+				</label>
+				<meter :id="questionOption.text.replace(/\s+/g, '')"
+					min="0"
+					:max="submissions.length"
+					:value="questionOption.count" />
+			</li>
+		</ol>
+		<ul v-else class="question-summary__text">
+			<li v-for="(textAnswer, index) in textAnswers"
+				:key="index">
+				<template v-if="index === 0">
+					<strong>{{ textAnswer }}</strong>
+				</template>
+				<template v-else>
+					{{ textAnswer }}
+				</template>
+			</li>
+		</ul>
+	</div>
+</template>
+
+<script>
+import answerTypes from '../../models/AnswerTypes'
+
+export default {
+	name: 'Summary',
+
+	props: {
+		submissions: {
+			type: Array,
+			required: true,
+		},
+		question: {
+			type: Object,
+			required: true,
+		},
+	},
+
+	data() {
+		return {
+			answerTypes,
+		}
+	},
+
+	computed: {
+		// For countable questions like multiple choice and checkboxes
+		questionOptions() {
+			const questionOptionsStats = []
+
+			// Also record 'No response'
+			questionOptionsStats.push({
+				'text': t('forms', 'No response'),
+				'count': 0,
+				'percentage': 0,
+			})
+
+			// Build list of question options
+			this.question.options.forEach(option => {
+				questionOptionsStats.push({
+					'text': option.text,
+					'count': 0,
+					'percentage': 0,
+				})
+			})
```

Comment:
You can use Array.map directly :)

```suggestion
			// Build list of question options
			const questionOptionsStats = this.question.options.map(option => ({
				text: option.text,
				count: 0,
				percentage: 0,
			}))

			// Also record 'No response'
			questionOptionsStats.unshift({
				text: t('forms', 'No response'),
				count: 0,
				percentage: 0,
			})
```

Reply:
Hehe, yep that’s the advanced stuff I didn’t get my head wrapped around yet. Thanks a lot, fixed! :)